code-challenges.test.js

```diff
@@ -15,35 +15,127 @@
 
-const colors1 = ["purple", "blue", "green", "yellow", "pink"]
-// Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
+describe("firstShuffle", () => {
+    const colors1 = ["purple", "blue", "green", "yellow", "pink"];
+    // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
+    const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+    // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+    it("takes in an array, removes the first item from the array and shuffles the remaining content", () => {
+      expect(["yellow", "blue", "pink", "green"]).toEqual(
+        expect.not.arrayContaining(colors1),
+        expect(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]).toEqual(
+            expect.not.arrayContaining(colors2),
+      ));
+    });
+  });
+
+//* RAN TEST (RESULTS BELOW): *//
+
+// PASS  ./code-challenges.test.js
+// firstShuffle
+//   ✓ takes in an array, removes the first item from the array and shuffles the remaining content (1 ms)
 
 // b) Create the function that makes the test pass.
 
+//Goal: Takes in an array, removes the first item from the array and shuffles the remaining content
+
+//Pseudocode:
+// Declare a function called firstShuffle. 
+// Set the parameter as an shuffleArray.
+// Use .shift to remove the first element of original array.
+// Use .map to put each element in an object type setup and give them a random sort key. 
+// Use .sort to produce an a new arrangement using the random key. 
+// Return expected output. 
+
+const firstShuffle = (shuffleArray) => {
+    return shuffleArray.shift() + shuffleArray.map(value => ({ value, sort: Math.random() })) + shuffleArray.sort((a, b) => a.sort - b.sort)
+    
+}
+
+//* RAN TEST (RESULTS BELOW): *//
+
+// (N/A - Nothing showed up when I ran the test, just said PASS but only as if I had the describe method aka Part A info only...it didn't register my logic.)
+
+// ** NOTES TO INSTRUCTOR **: I wasn't sure if I used the correct describe method setup based on the linked resource but I made my best guest and received a PASS for it. However, when it came to the second part of the question (the logic) as you can read above in the "RAN TEST" section it didn't register in Jest, it doesn't run the test for my logic for some reason. Any clarity on that? Regarding the logic, I wasn't sure if I should use Math.floor(Math.random() as I thought they may be mostly used for integers but I wasn't sure how else to randomize the elements of the array without it and my research wasn't providing many alternatives either. Side question should I have used .slice? 
+
```

Comment:
Hi Steven - you were on the right track, just got a few things mixed up.  In the first part of your expect statement, you want to call your method and pass in the array ===> expect(firstShuffle(colors1)).   Then in the second part, you expect to get an array containing the remaining colors (after first element is removed) ===> (expect.arrayContaining(the expected outcome)).  final:        expect(shuffle(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"])). Hope this helps!

Reply:
Thank you, that clears up some things!